inst/qml/NonParametricSurvivalAnalysis.qml

```diff
@@ -0,0 +1,231 @@
+import QtQuick.Layouts	1.3
+import JASP.Controls	1.0
+import JASP.Widgets		1.0
+import JASP				1.0
+
+Form
+{
+	VariablesForm
+	{
+		AvailableVariablesList
+		{
+			name: "allVariablesList"
+		}
+
+		AssignedVariablesList
+		{
+			name:				"timeToEvent"
+			title:				qsTr("Time to Event")
+			suggestedColumns:	["scale"]
+			singleVariable:		true
+		}
+
+		AssignedVariablesList
+		{
+			id:					eventStatusId
+			name:				"eventStatus"
+			title:				qsTr("Event Status")
+			suggestedColumns:	["nominal"]
+			singleVariable:		true
+		}
+
+		DropDown
+		{
+			name:				"eventIndicator"
+			label:				qsTr("Event Indicator")
+			source:				[{name: "eventStatus", use: "levels"}]
+			onCountChanged:		currentIndex = 1
+		}
+
+		AssignedVariablesList
+		{
+			name:			 	"factors"
+			title:			 	qsTr("Factors")
+			allowedColumns:		["ordinal", "nominal", "nominalText"]
+		}
+	}
+
+	Group
+	{
+
+		CheckBox
+		{
+			name:	"tests"
+			label:	qsTr("Tests")
+
+			CheckBox
+			{
+				name:		"testsLogRank"
+				label:		qsTr("Log-rank (Mantel-Haenszel)")
+				checked:	true
+			}
+
+			CheckBox
+			{
+				name:		"testsPetoAndPeto"
+				label:		qsTr("Peto and Peto")
+				checked:	true
+			}
+
+			CheckBox
+			{
+				name:		"testsFlemmingHarrington"
+				label:		qsTr("Flemming-Harrington")
+
+				DoubleField
+				{
+					name:			"testsFlemmingHarringtonRho"
+					label:			qsTr("Rho")
+					defaultValue:	0.5
+					min:			0
+					max:			1
+				}
+			}
+		}
+
+		CheckBox
+		{
+			name:	"lifeTable"
+			label:	qsTr("Life table")
+
+			DropDown
+			{
+				name:		"lifeTableStepsType"
+				id:			lifeTableStepsType
+				label:		qsTr("Steps type")
+				values:
+				[
+					{ label: qsTr("Default"),		value: "default"},
+					{ label: qsTr("Quantilies"),	value: "quantiles"},
+					{ label: qsTr("Fixed size"),	value: "fixedSize"}
+				]
+			}
+
+			IntegerField
+			{
+				name:			"lifeTableStepsNumber"
+				label:			qsTr("Number")
+				defaultValue:	10
+				visible:		lifeTableStepsType.value == "quantiles"
+			}
+
+			CheckBox
+			{
+				name:		"lifeTableRoundSteps"
+				label:		qsTr("Round steps")
+				checked:	true
+				visible:	lifeTableStepsType.value == "quantiles"
+			}
+
+			DoubleField
+			{
+				name:			"lifeTableStepsFrom"
+				id:				lifeTableStepsFrom
+				label:			qsTr("From")
+				defaultValue:	0
+				max:			lifeTableStepsTo.value
+				visible:		lifeTableStepsType.value == "fixedSize"
+			}
+
+			DoubleField
+			{
+				name:			"lifeTableStepsSize"
+				label:			qsTr("Size")
+				defaultValue:	1
+				// max:			lifeTableStepsTo.value // TODO: enable once max is data dependent
+				visible:		lifeTableStepsType.value == "fixedSize"
+			}
+
+			DoubleField
+			{
+				name:			"lifeTableStepsTo"
+				id:				lifeTableStepsTo
+				label:			qsTr("To")
+				defaultValue:	0
+				min:			lifeTableStepsFrom.value
+				visible:		lifeTableStepsType.value == "fixedSize"
+			}
+		}
+	}
+
+	CheckBox
+	{
+		name:	"survivalCurvePlot"
+		label:	qsTr("Survival curve plot")
```

Comment:
The risk table, cumulative events, censoring plots, etc. are nested under survival curve plot despite generating independent plots. Is that on purpose? Could we put it under a `Group`/`Section` titled "Plots" and have them as separate options?

Reply:
Yeah, this one is actually on purpose -- the final figure should be a one panel, but we were having trouble aligning the x-axis etc, properly. Hopefully, we could add a checkbox to the option that merges them together (as supposed to) in the next version.